common/services/user.js

```diff
@@ -43,9 +44,16 @@ function getAuthGroups(token) {
 function getAuthLocations(token) {
   const { authorities = [] } = decodeAccessToken(token)
 
-  return getAuthGroups(token).then(groups => {
+  return getAuthGroups(token).then(async groups => {
     if (authorities.includes('ROLE_PECS_SUPPLIER')) {
-      return getLocationsBySupplierId(groups[0])
+      try {
+        const supplierKey = groups[0].toLowerCase()
+        const supplier = await getSupplierByKey(supplierKey)
+
+        return getLocationsBySupplierId(supplier.id)
+      } catch (err) {
+        return Promise.resolve([])
```

Comment:
do we want the error to be silent here?

Reply:
So I kind of thought, yes. Mainly because in all the error cases in this instance are mostly expected and we then just want to return an empty array of locations for the app to handle.

Errors would be:
* user has no groups
   * `try/catch` will error trying to convert group to lowercase and return empty array
* user has groups, but no supplier found
  * `try/catch` will catch `404` response from `getSupplierByKey` and return empty array
* user has groups, supplier is found, but no locations
  * `catch` will not be run but `getLocationsBySupplierId` will return an empty array

Reply:
if `getSupplierByKey` actually errors though we won't know right? this will just return an empty array.

It feels like we could cater for all these things with `catch` maybe?

```
try {
  const supplierKey = groups[0].toLowerCase()
  const supplier = await getSupplierByKey(supplierKey)

  return getLocationsBySupplierId(supplier.id)
} catch (err instanceof TypeError) { // no elements in array
  return Promise.resolve([])
} catch (err.code === '404') {
  return Promise.resolve([])
} catch (err) { // all otehr errors
  logger.error(error)
}
```